### PR TITLE
[DataTable]: sticky rows

### PR DESCRIPTION
**Pull request by Kuznietsov**
## In progress...

### CODE REVIEW

File: uui/components/tables/DataRowsContainer/DataRows.tsx [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+            </div>
+            {childrenRows.length > 0 && (
+                <div>
+                    <DataRows rows={ childrenRows } renderRow={ renderRow } top={ childrenPinnedTop } />
+                </div>
+            )}
+        </div>
+    );
+}
+export function DataRows<TItem, TId>({
```
Comment (jakobz) [inline uui/components/tables/DataRowsContainer/DataRows.tsx:40]: We need another name for this component. With just 'DataRows', I'm getting confused with DataRowsProps interface - it clashes with DataRowProps (with singular 'Row') - which have a different meaning.

File: uui/components/tables/DataRowsContainer/DataRows.tsx [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+}
+
+function DataRowGroup<TItem, TId>({
+    row,
+    childrenRows,
+    renderRow,
+    top = 1,
+}: GroupProps<TItem, TId>) {
+    const rowRef = useRef<HTMLDivElement>();
+    const childrenPinnedTop = row.isPinned ? (top + (rowRef.current?.clientHeight ?? 0)) : top;
```
Comment (jakobz) [inline uui/components/tables/DataRowsContainer/DataRows.tsx:26]: As we'll get ref.current only after rendering, we won't get correct values for 'top' on the first rendering. However, this works, as we have few re-renders before some rows get stuck. Correct?

Comment (Kuznietsov) [reply uui/components/tables/DataRowsContainer/DataRows.tsx:26]: Yes, that's correct.

File: uui/components/tables/DataRowsContainer/DataRows.tsx (hunk 1 above)
Comment (Kuznietsov) [reply uui/components/tables/DataRowsContainer/DataRows.tsx]: Done.

**Comment by Kuznietsov** [comment]
Merged into feature/sticky-rows-with-scroll-to manually.

**Pull request closed without merging**